openmp/runtime/src/kmp_platform.h

```diff
@@ -76,14 +77,20 @@
 #define KMP_OS_SOLARIS 1
 #endif
 
+#if (defined __wasi__) || (defined __EMSCRIPTEN__)
```

Comment:
@abrown 
This is conflating wasi and the emscripten toolchain. They are two different cases where emscripten mimics linux/posix system by statically linking musl into the wasm file.

The changes in this PR will not allow openmp to compile under emscripten, the changes here will
https://github.com/llvm/llvm-project/pull/95169

Reply:
Good finds, both of them. Let's discuss in #95169.